### PR TITLE
Support machine image versions to specify constraint for supported kubelet versions

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/kind enhancement
/kind api-change

**What this PR does / why we need it**:
Kubernetes 1.26 removed support for CRI v1alpha2 - see https://kubernetes.io/blog/2022/12/09/kubernetes-v1-26-release/#cri-v1alpha2-removed.
It means that kubelet@1.26 is only compatible with containerd versions >= 1.6.0. There are out there still machine image versions using containerd < 1.6.0.
Currently in the Gardener API there is no way to describe such constraints. This PR introduces support for specifying kubelet version constraint per machine image version.
With this feature, gardener operators could prepare the CloudProfiles for 1.26 as follows:
```yaml
apiVersion: core.gardener.cloud/v1beta1
kind: CloudProfile
# ...
spec:
  machineImages:
  - name: gardenlinux
    versions:
    - version: 934.1.0 # comes with containerd@1.6.8
    - version: 576.12.0 # comes with containerd@1.5.13
      kubeletVersionConstraint: "< 1.26"
```

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6773

**Release note**:

```feature operator
The Gardener API Server now supports machine image versions to specify constraints for the supported kubelet versions. This is a feature needed in the context of Kubernetes 1.26 adoption. kubelet@1.26 removed support CRI v1alpha2. Hence, kubelet@1.26 is only compatible with containerd versions >= 1.6.0. Using this feature gardener operators can specify that a given machine image version with containerd < 1.6.0 is only usable by kubelet < 1.26.
```

### CODE REVIEW

File: plugin/pkg/shoot/validator/admission.go [hunk 1]
```diff
@@ -781,8 +788,16 @@ func (c *validationContext) validateProvider(a admission.Attributes) field.Error
 				detail := fmt.Sprintf("machine type '%s' does not support CPU architecture '%s' or is unavailable in at least one zone, supported machine types are: %+v", worker.Machine.Type, *worker.Machine.Architecture, validMachineTypes)
 				allErrs = append(allErrs, field.Invalid(idxPath.Child("machine", "type"), worker.Machine.Type, detail))
 			}
-			if ok, validMachineImages := validateMachineImagesConstraints(c.cloudProfile.Spec.MachineImages, worker.Machine, oldWorker.Machine); !ok {
-				detail := fmt.Sprintf("machine image '%s' does not support CPU architecture '%s' or is expired, valid machine images are: %+v", worker.Machine.Image, *worker.Machine.Architecture, validMachineImages)
+
+			kubeletVersion, err := helper.CalculateEffectiveKubernetesVersion(controlPlaneVersion, worker.Kubernetes)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(idxPath, worker.Kubernetes, "cannot determine effective Kubernetes version for worker pool"))
```
Comment (acumino) [inline plugin/pkg/shoot/validator/admission.go]: The error can occur only if version is specified.
```suggestion
				allErrs = append(allErrs, field.Invalid(idxPath.Child("version"), worker.Kubernetes.Version, "cannot determine effective Kubernetes version for worker pool"))
```

File: pkg/controllermanager/controller/shoot/maintenance/reconciler_test.go [hunk 2]
```diff
@@ -363,6 +366,29 @@ var _ = Describe("Shoot Maintenance", func() {
 			assertWorkerMachineImageVersion(&shoot.Spec.Provider.Workers[2], "CoreOs", "1.1.1")
 		})
 
+		It("should determine that the shoot worker machine images must not be maintained - found no machineImageVersion with matching kubeletVersionConstraint constraint (control plane K8s version)", func() {
```
Comment (acumino) [inline pkg/controllermanager/controller/shoot/maintenance/reconciler_test.go:369]: Can we add one more test case where machineImageVersion is upgraded?

File: plugin/pkg/shoot/validator/admission.go [hunk 3]
```diff
@@ -791,7 +791,7 @@ func (c *validationContext) validateProvider(a admission.Attributes) field.Error
 
 			kubeletVersion, err := helper.CalculateEffectiveKubernetesVersion(controlPlaneVersion, worker.Kubernetes)
 			if err != nil {
-				allErrs = append(allErrs, field.Invalid(idxPath, worker.Kubernetes, "cannot determine effective Kubernetes version for worker pool"))
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("kubernetes").Child("version"), worker.Kubernetes.Version, "cannot determine effective Kubernetes version for worker pool"))
```
Comment (acumino) [inline plugin/pkg/shoot/validator/admission.go]: Nit: This looks better.
```suggestion
				allErrs = append(allErrs, field.Invalid(idxPath.Child("kubernetes", "version"), worker.Kubernetes.Version, "cannot determine effective Kubernetes version for worker pool"))
```

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by rfranzke — COMMENTED** [review]
/lgtm

**Pull request merged**